### PR TITLE
feat: enable pebble

### PR DESCRIPTION
**Pull request by faddat**
- support pebble
- bump pyroscope

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Comment by faddat** [comment]
@coderabbitai review

@coderabbitai & reviewers - to finish off this PR, there's a second PR on the celestia-core repository, here:

* https://github.com/celestiaorg/celestia-core/pull/1237

The celestia-core PR allows celestia-core to use go 1.22 and so it needed to update some pyroscope dependencies. 

There's also a PR to cometbft-db here:

* https://github.com/cometbft/cometbft-db/pull/135

Once the cometbft-db PR is merged, I'll be able to use it without the replace statement in the celestia-core PR, and in this PR, too.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/celestiaorg/celestia-app
 
-go 1.21.1
+go 1.22
+
+toolchain go1.22.0
```
Comment (evan-forbes) [inline go.mod:5]: ideally we would do this in a separate PR, is this required for the other changes here? if so, then we will likely need to create a different PR that also updates all the CI to use the latest go version

File: go.mod [hunk 2]
```diff
@@ -223,5 +233,6 @@ replace (
 	github.com/cosmos/cosmos-sdk => github.com/celestiaorg/cosmos-sdk v1.18.3-sdk-v0.46.14
 	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
 	github.com/syndtr/goleveldb => github.com/syndtr/goleveldb v1.0.1-0.20210819022825-2ae1ddf74ef7
-	github.com/tendermint/tendermint => github.com/celestiaorg/celestia-core v1.32.0-tm-v0.34.29
+	github.com/tendermint/tendermint => github.com/faddat/celestia-core v1.29.0-tm-v0.34.29.0.20240221225107-acff1e186d7b
+	github.com/tendermint/tm-db => github.com/faddat/tm-db v0.6.6-0.20240221221500-fc0fbd3cc893
```
Comment (evan-forbes) [inline go.mod:237]: does this what are the other consequences of this? could this be upstreamed or is this also blocked by an upstream PR?

**Review by evan-forbes — COMMENTED** [review]
thanks for this @faddat

**Comment by rootulp** [comment]
Closing b/c this doesn't appear to be actively worked on. Please re-open if I misinterpreted.

**Pull request closed without merging**